lib/infrastructure/plan/plan_data_source.dart

```diff
@@ -40,4 +46,58 @@ class PlanDataSource extends _$PlanDataSource implements PlanRepository {
     }
     return;
   }
+
+  Future<List<String>> getBookmarkedPlanIds() async {
+    final userData = fireauth.currentUser;
+    if (userData == null) {
+      return [];
+    }
+    final snapshot =
+        await firestore.collection('users').doc(userData.uid).get();
+    if (!snapshot.exists ||
+        !snapshot.data()!.containsKey('bookmarkedPlanIds')) {
+      return [];
+    }
+    final bookmarkedPlanIds = snapshot.data()!['bookmarkedPlanIds'];
+    if (bookmarkedPlanIds == null || bookmarkedPlanIds is! List<dynamic>) {
+      return [];
+    }
+    return List<String>.from(bookmarkedPlanIds);
+  }
+
+  Future<Plan> getPlanData({required String planId}) async {
+    final topics = <Topic>[];
+    final snapshot =
+        await firestore.collection('PopularPlans').doc(planId).get();
+    final data = snapshot.data();
+    if (data == null) {
+      return const Plan(
+        title: '',
+        description: '',
+        thumbnailUrl: '',
+        topics: [],
+      );
+    }
+    for (final topicId in data['topics'] as List) {
+      final topic = await topicsorce.getTopicData(topicId: topicId.toString());
+      topics.add(Topic(name: topic.name, thumbnailUrl: ''));
```

Comment:
getTopicDataメソッドを使用するとそのままTopic型のデータが返ってくるので、そのまま追加しよう
```suggestion
      final topic = await topicsorce.getTopicData(topicId: topicId.toString());
      topics.add(topic);
```